plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java

```diff
@@ -0,0 +1,147 @@
+        this.actionFilters = actionFilters;
+        this.transportService = transportService;
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<MLUpdateModelGroupResponse> listener) {
+        MLUpdateModelGroupRequest updateModelGroupRequest = MLUpdateModelGroupRequest.fromActionRequest(request);
+        MLUpdateModelGroupInput updateModelGroupInput = updateModelGroupRequest.getUpdateModelGroupInput();
+
+        String modelGroupId = updateModelGroupInput.getModelGroupID();
+        GetRequest getModelGroupRequest = new GetRequest(ML_MODEL_GROUP_INDEX).id(modelGroupId);
+
+        if (modelGroupId != null) {
+            User user = RestActionUtils.getUserContext(client);
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                client.get(getModelGroupRequest, ActionListener.wrap(modelGroup -> {
+                    if (modelGroup.isExists()) {
+                        Map<String, Object> source = modelGroup.getSourceAsMap();
+                        Map<String, Object> owner = (Map) source.get(MLModelGroup.OWNER);
+                        String access = (String) source.get(MLModelGroup.ACCESS);
+
+                        if (SecurityUtils.isAdmin(user) || access == "public") {
+                            updateModelGroup(modelGroupId, source, updateModelGroupInput, listener);
+                        } else if (user.getName().equals(owner.get("name"))) {
+                            if (updateModelGroupInput.getBackendRoles() != null) {
+
+                                List<String> backendRoles = user.getBackendRoles() != null ? user.getBackendRoles() : ImmutableList.of();
+
+                                if (updateModelGroupInput.getBackendRoles().size() == 0) {
+                                    updateModelGroupInput.setBackendRoles(backendRoles);
+                                }
+
+                                Boolean isRolePresent = updateModelGroupInput
+                                    .getBackendRoles()
+                                    .stream()
+                                    .allMatch(user.getBackendRoles().stream().collect(Collectors.toSet())::contains);
+
+                                if (!isRolePresent) {
+                                    log.error("Invalid Backend Roles provided in the input");
+                                    throw new IllegalArgumentException("Invalid Backend Roles provided in the input");
+                                }
+                            }
+
+                            updateModelGroup(modelGroupId, source, updateModelGroupInput, listener);
+                        } else {
+                            throw new IllegalArgumentException("User doesn't have valid privilege to perform this operation");
```

Comment:
Only the model is public or user is admin/owner has the permission to update the model? Others with a subset backend roles can not update the model, correct?

Reply:
actually even if the model group is public, we should not let other users change back-end roles field. For that reason, I am refactoring the code to only allow owner/admin to update a model group